wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubFactoryParametersResolver.java

```diff
@@ -91,13 +91,15 @@ public FactoryDto createFactory(@NotNull final Map<String, String> factoryParame
       throws BadRequestException, ServerException {
 
     // no need to check null value of url parameter as accept() method has performed the check
-    final GithubUrl githubUrl = githubUrlParser.parse(factoryParameters.get(URL_PARAMETER_NAME));
+    final GithubUrl githubUrl = githubUrlParser.parse(factoryParameters.remove(URL_PARAMETER_NAME));
```

Comment:
I have concerns if it's the right way to propagate every unknown query parameter as devfile override.
It's kind of change the current behavior, now we just ignore such.

Maybe we could solve this issue in the way how WorkspaceService#create. Then it could be look like the folowing:
`factory?url=https://github.com/eclipse/che&override=projects.che.source.branch:featureBranch&override=projects.che.name:my-che`.
From one point - url becomes a bit complicated comparing to the implemented way `factory?url=https://github.com/eclipse/che&projects.che.source.branch=featureBranch&projects.che.name=my-che`
But from another view - overrides become more explicit, and I consider it as good stuff.

Reply:
I'm not sure we need to do that. We have control over both the query parameter naming and devfile schema so we can ensure there are no clashes.

Reply:
It's not only about clashes but about transparency, with `override` query parameter in REST API docs (for example swagger) you're able to document the format of such parameter.

One more plus one to `override` query param: it's just less error-prone.
We can imagine that GithubFactoryResolver supports param - `branch`, but it's not supported by DefaultFactoryResolver. But then all implementations should we aware about all possible parameters, and DefaultFactoryResolver must remove it from the map before propagating to further.

Also, should `FactoryResolver` implementation carry about preparing overrides from factoryParameters at all? Maybe it would be easier to prepare them once at FactoryService level and then propagate?
```
interface FactoryResolver {
  /**
  * @param factoryParameters: implementation specific parameters that should be used for devfile evaluating
  * @param overrides: overrides that should be applied to the devfile resolved by implementation
  */
  public boolean accept(@NotNull final Map<String, String> factoryParameters, final Map<String, String> overrides);
}
```

Reply:
Refactored to use prefixed params